project/Build.scala

```diff
@@ -638,6 +638,12 @@ object Build {
         testDefinitionsFile +: (jsExecutionFiles in Test).value
       },
 
+      jsExecutionFiles in Test := {
+        val resourceDir = (resourceDirectory in (testSuite, Test)).value
+        val f = FileVirtualJSFile(resourceDir / "ScalaJSDefinedTestNatives.js")
+        f +: (jsExecutionFiles in Test).value
+      },
```

Comment:
I, er, haven't. You mean something like this, I guess:
```scala
def testJSExecutionFilesSetting: Setting[_] = {
  jsExecutionFiles in Test := {
    val resourceDir = (resourceDirectory in (testSuite, Test)).value
    val f = FileVirtualJSFile(resourceDir / "ScalaJSDefinedTestNatives.js")
    f +: (jsExecutionFiles in Test).value
  }
}
```
and then use it both in `testSuite.settings(...)` and `toolsJS.settings(...)`?